src/runtime_src/core/tools/common/XBUtilities.cpp

```diff
@@ -386,13 +386,24 @@ XBUtilities::collect_devices( const std::set<std::string> &_deviceBDFs,
 }
 
 xrt_core::device_collection
-XBUtilities::collect_devices( const std::string& _devices, // comma separated no space
-                              bool _inUserDomain )
+XBUtilities::
+collect_devices(const std::vector<std::string>& _devices, bool _inUserDomain)
```

Comment:
Should this be exposed? This looks like a helper function to the following func

Reply:
Not sure what you mean by exposed?  Some commands build a device list as a comma separated string, others build it as a vector of string.  I don't know why there is this inconsistency, but I am hiding it in XBU so commands look the same regardless of device list format.

Reply:
I didn't realize that there was an inconsistency. My guess is it's happening because of how device gets initialized and captured by the boost program_options: vector captured using multitoken vs capturing a simple comma-separated string. 

We should work on removing: `collect_devices( const std::string& _devices, bool _inUserDomain )` and having only `collect_devices(const std::vector<std::string>& _devices, bool _inUserDomain)`. I can look into this once I have a chance. Created a ticket: XRT-699